### PR TITLE
Adding Ice Driver Support

### LINKED ISSUE #9: ice Support?
Hello,

Is there any thought to add support for ice driver PFs? 

When running the exporter for ice driver VFs I get:
```
# ./sriov-exporter -collector.vfstats -web.listen-address :9102 
2021/12/28 05:38:27 The vfstats collector is enabled
2021/12/28 05:38:27 listening on :9102
2021/12/28 05:38:32 No stats reader available for Physical Function eth4
2021/12/28 05:38:32 No stats reader available for Physical Function eth5
...

# ip link
...
64: eth4: <BROADCAST,MULTICAST,PROMISC,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether b4:96:91:b4:4a:78 brd ff:ff:ff:ff:ff:ff
    vf 0     link/ether 00:11:22:33:11:11 brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state enable, trust on
    vf 1     link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state enable, trust on
65: eth5: <BROADCAST,MULTICAST,PROMISC,UP,LOWER_UP> mtu 1500 qdisc mq state UP mode DEFAULT group default qlen 1000
    link/ether b4:96:91:b4:4a:79 brd ff:ff:ff:ff:ff:ff
    vf 0     link/ether 00:11:22:33:11:22 brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state enable, trust on
    vf 1     link/ether 00:00:00:00:00:00 brd ff:ff:ff:ff:ff:ff, spoof checking off, link-state enable, trust on
...

# driverctl -v list-devices | grep -i eth
...
0000:4b:00.0 ice [*] (Ethernet Controller E810-C for QSFP (Ethernet Network Adapter E810-C-Q2))
0000:4b:00.1 ice [*] (Ethernet Controller E810-C for QSFP (Ethernet Network Adapter E810-C-Q2))
0000:4b:01.0 vfio-pci [*] (Ethernet Adaptive Virtual Function)
0000:4b:01.1 vfio-pci [*] (Ethernet Adaptive Virtual Function)
0000:4b:11.0 vfio-pci [*] (Ethernet Adaptive Virtual Function)
0000:4b:11.1 vfio-pci [*] (Ethernet Adaptive Virtual Function)
...
```

Thank you,
Vangelis

### PR DESCRIPTION
**Pull request by Eoghan1232**
Ice driver support added
redundant variables removed
new utils pkg added
File Path cleaning implemented
Hadolint errors handled
Golangci-lint errors handled

also fixes #9 

signed off : eoghan.russell@intel.com & eoghan.lawless@intel.com

### CODE REVIEW

**Comment by Eoghan1232** [comment]
@esiperu-est can you take a look please

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package utils
+
+import (
+	"log"
+	"path/filepath"
+)
+
+func VerifyPath(path *string) {
```
Comment (zeeke) [inline pkg/utils/utils.go:8]: nit: What about renaming to `ResolvePath` or `ResolveSymlinks`? "Verify" seems to be a passive action, while this function changes the value of its input parameter.

Comment (Eoghan1232) [reply pkg/utils/utils.go]: Agreed. Done.

**Comment by zeeke** [comment]
LGTM

**Pull request merged**